modules/table/src/main/java/org/apache/ignite/internal/table/distributed/SortedIndexLocker.java

```diff
@@ -155,15 +154,15 @@ public CompletableFuture<Lock> locksForInsert(UUID txId, BinaryRow tableRow, Row
 
         BinaryTuplePrefix prefix = BinaryTuplePrefix.fromBinaryTuple(key);
 
-        // find next key
-        Cursor<IndexRow> cursor = storage.scan(prefix, null, SortedIndexStorage.GREATER);
+        // Find next key.
+        PeekCursor<IndexRow> cursor = storage.scan(prefix, null, SortedIndexStorage.GREATER);
 
-        BinaryTuple nextKey;
-        if (cursor.hasNext()) {
-            nextKey = cursor.next().indexColumns();
-        } else { // otherwise INF
-            nextKey = POSITIVE_INF;
-        }
+        //noinspection ResultOfMethodCallIgnored
+        cursor.hasNext();
+
+        // "peek" works the same as "next" when "hasNext" is called. The only difference is that it returns null instead of throwing
+        // an exception, if nothing is found.
+        BinaryTuple nextKey = indexKey(cursor.peek());
```

Comment:
The code 'after' seems to be equivalent to the code 'before', but the code 'before' was arguably more straightforward. Are you sure this actually improves readability?

Reply:
Old code was a scenario copy-pasted twice, I removed the copy-paste.
What else is equivalent? Replacement of "hasNext()" with "peek() != null" is not an equivalent change, it's a fix

Reply:
My comment is not relevant anymore, please ignore it